scripts/ioc_deploy.py

```diff
@@ -0,0 +1,327 @@
+#!/usr/bin/python3
+"""
+ioc-deploy is a script for building and deploying ioc tags from github.
+It will create a shallow clone of your IOC in the standard release area at the correct path and "make" it.
```

Comment:
It might be nice to include an explicit example of the most common use case and expected result. Something like:
`ioc-deploy --name ioc-foo-bar -r R1.0.0`
will clone `ioc-foo-bar` tag `R1.0.0` to `/cds/group/pcds/epics/ioc/common/foo/bar/R1.0.0` and run `make` using the currently set EPICS environment variables.

Reply:
I'll do this prior to merge, it's important to document these fully.

Reply:
This is now included in the docstring and help text